chat/documents.py

```diff
@@ -63,41 +56,49 @@ def save(self, *args, **kwargs):
 
 class Message(Document):
     """
-    Represents a single chat message in a room.
+    Represents a chat message stored in MongoDB.
 
     Attributes:
-        room (Room): Reference to the room where the message belongs.
-        sender (User): Reference to the user who sent the message.
-        text (str): Message content, 1–1000 characters.
-        timestamp (datetime): Time when the message was created (UTC).
-        is_read (bool): Indicates whether the message has been read.
+        room (Room): Reference to the chat room where the message was sent.
+        sender_id (str): ID of the user who sent the message (from PostgreSQL).
+        receiver_id (str, optional): ID of the recipient in private chats. None for group messages.
+        text (str): The content of the message, between 1 and 1000 characters.
+        timestamp (datetime): The time when the message was created (UTC).
+        is_read (bool): Indicates whether the message has been read by the recipient(s).
     """
-
     room = ReferenceField(Room, required=True, reverse_delete_rule=CASCADE)
-    sender = ReferenceField('UserDocument', required=True)
-    text = StringField(required=True, min_length=MIN_MESSAGE_LENGTH, max_length=MAX_MESSAGE_LENGTH)
+    sender_id = StringField(required=True)
+    receiver_id = StringField(required=False)
+    text = StringField(required=True, min_length=1, max_length=1000)
     timestamp = DateTimeField(default=lambda: datetime.now(timezone.utc))
     is_read = BooleanField(default=False)
 
     def clean(self):
         """
-        Validate message data before saving:
-        - Ensure the message has a valid room.
-        - Ensure sender is persisted and part of the room.
-        - Ensure message doesn't contain forbidden words.
-        - Block messages that appear spammy (repeated characters).
+        Validation rules:
+        - Room must exist.
+        - Sender must be in room participants.
+        - If private chat: receiver must be set and also in participants.
+        - In group chat: receiver can be None (message for everyone).
         """
         if not self.room:
             raise ValidationError("Message must belong to a valid room.")
 
-        if not self.sender or not self.sender.id:
-            raise ValidationError("Sender must be a persisted user.")
-
-        participant_ids = {u.id for u in self.room.participants}
-        if self.sender.id not in participant_ids:
+        if self.sender_id not in self.room.participants:
             raise ValidationError("Sender must be a participant of the room.")
 
-        if not self.text:
+        if self.room.is_group:
+            if self.receiver_id and self.receiver_id not in self.room.participants:
+                raise ValidationError("Receiver must be a participant of the group.")
```

Comment:
The receiver validation for group messages runs even if receiver_id is None. Add an explicit None check before validating receiver_id.
```
if self.room.is_group and self.receiver_id is not None:
    if self.receiver_id not in self.room.participants:
        raise ValidationError("Receiver must be a participant of the group.")
```